APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m

```diff
@@ -0,0 +1,293 @@
+    
+    [self writeDictionary:dict toPath:[self baseTrackingFilePath]];
+}
+
+- (CLLocation*)baseTrackingLocation
+{
+    if (!_baseTrackingLocation)
+    {
+        if (self.folder)
+        {
+            if ([[NSFileManager defaultManager] fileExistsAtPath:[self baseTrackingFilePath]])
+            {
+                NSError*    error       = nil;
+                NSString*   jsonString  = [NSString stringWithContentsOfFile:[self baseTrackingFilePath]
+                                                                    encoding:NSUTF8StringEncoding
+                                                                       error:&error];
+                
+                if (!jsonString)
+                {
+                    if (error)
+                    {
+                        APCLogError2(error);
+                    }
+                }
+                else
+                {
+                    NSDictionary* dict      = nil;
+                
+                    dict                    = [NSDictionary dictionaryWithJSONString:jsonString];
+                    _baseTrackingLocation   = [[CLLocation alloc] initWithLatitude:[dict[kLat] doubleValue]
+                                                                         longitude:[dict[kLon] doubleValue]];
+                }
+            }
+        }
+    }
+    
+    return _baseTrackingLocation;
+}
+
+- (void)setMostRecentUpdatedLocation:(CLLocation*)mostRecentUpdatedLocation
+{
+    _mostRecentUpdatedLocation = mostRecentUpdatedLocation;
+    
+    NSDictionary * dict = @{kLat : @(mostRecentUpdatedLocation.coordinate.latitude), kLon : @(mostRecentUpdatedLocation.coordinate.longitude)};
+    
+    [self writeDictionary:dict toPath:[self recentLocationFilePath]];
+}
+
+- (CLLocation*)mostRecentUpdatedLocation
+{
+    if (!_mostRecentUpdatedLocation)
+    {
+        if (self.folder)
+        {
+            if ([[NSFileManager defaultManager] fileExistsAtPath:[self recentLocationFilePath]])
+            {
+                NSError*    error       = nil;
+                NSString*   jsonString  = [NSString stringWithContentsOfFile:[self recentLocationFilePath]
+                                                                    encoding:NSUTF8StringEncoding
+                                                                       error:&error];
+                
+                if (!jsonString)
+                {
+                    if (error)
+                    {
+                        APCLogError2(error);
+                    }
+                }
+                else
+                {
+                    NSDictionary* dict      = nil;
+
+                    dict                        = [NSDictionary dictionaryWithJSONString:jsonString];
+                    _mostRecentUpdatedLocation  = [[CLLocation alloc] initWithLatitude:[dict[kLat] doubleValue]
```

Comment:
This is an almost an exact copy of the code for baseTrackingLocation, with a different path. Needs factoring out into a single method.